claim/utils.py

```diff
@@ -46,3 +47,26 @@ def process_items_relations(user, claim, items):
 
 def process_services_relations(user, claim, services):
     return process_child_relation(user, services, claim.id, claim.services, service_create_hook)
+
+
+def __autogenerate_nepali_claim_code():
+    code_length = ClaimConfig.claim_autogenerate_code_length
+    prefix = __get_current_nepali_fiscal_year_code()
+    last_claim = Claim.objects.filter(validity_to__isnull=True, code__icontains=prefix)
+    code = 0
+    if last_claim:
+        code = int(last_claim.latest('code').code[-code_length:])
+    return prefix + str(code+1).zfill(code_length)
+
+
+def __get_current_nepali_fiscal_year_code():
+    import nepali_datetime
```

Comment:
I think this import can be moved outside function

Reply:
I wanted to separate this in that place - the issue is that currently I had to add this library to the general requirements.txt file, but it would make more sense to design a system to handle implementation specific dependencies. Tanzanian implementation for instance should not be affected by this import and I believe it would be safer to keep it here.

Reply:
Thanks, with this explanation keeping it inside function is fine. As for the requirements.txt - this library could be added to setup.py. For the time being I think that keeping Nepali generator in claims.py is ok, however in the long term ideally it would be added using external module and config.